### PR TITLE
Euclidean algorithm: Output standardization

### PR DESCRIPTION
**Pull request by stormofice**
The following implementations were not fully standardized:

Implementations which seem to contain errors or do not run:
- clojure
- factor (can't get to run either [online](https://tio.run/#factor) or locally)
- smalltalk (apparently this is written in the Pharo dialect so smalltalk (as per the original PR #454), but it still does not work)
- viml (implementation seems kind of wrong, at least when I ran it [online](https://tio.run/#v))

No need to and/or not standardizable:
- emojicode (esolang)
- lolcode (esolang)
- matlab (output misses the newline after `[#]`, but as there are no plans in automatically validating it, I think this can be ignored)
- piet (graphical language)
- scratch (mostly graphical language)
- whitespace (esolang)

Two implementations had bugs (fixed now):
- Javascript
- Scala

### CODE REVIEW

**Comment by ntindle** [comment]
#878 has the scheme fix

**Comment by stormofice** [comment]
The languages which I standardized already can be reviewed, however it would be good to gather some more implementations before the final review.

**Comment by stormofice** [comment]
I am done with all the implementations which can be standardized properly (excluding esoteric and graphical languages).

Clojure / Factor / Smalltalk / viml were not standardized, as I could not figure out how to run them. I suspect that they either contain syntactic/semantic errors or that they do not run/build with newer versions of their respective programming language.

Apart from these outliers everything should be ready for review.

Currently there do not seem to be any reviewers active, which could take a look at the broken implementations.

However I *really* do want to stay back from merging this until those aforementioned implementations are fixed or removed, as just leaving them out would be against the original intent of this standardization work.

File: contents/euclidean_algorithm/code/scala/euclidean.scala [hunk 1]
```diff
@@ -3,8 +3,8 @@ object Euclid {
   def euclid_sub(a: Int, b: Int): Int =
     (Math.abs(a), Math.abs(b)) match {
       case (0, _) | (_, 0) => 0
-      case (x, y) if x < y => euclid(x, y - x)
-      case (x, y) if x > y => euclid(x - y, y)
+      case (x, y) if x < y => euclid_sub(x, y - x)
```
Comment (leios) [inline contents/euclidean_algorithm/code/scala/euclidean.scala:6]: Wait, how did this code run before?

File: contents/euclidean_algorithm/code/fortran/euclidean.f90 [hunk 2]
```diff
@@ -38,12 +38,18 @@ PROGRAM euclidean
...
-    WRITE(*,*) 'Modulus method:     GCD is: ', euclid_mod(a, b)
+    WRITE(*,'(a)') '[#]'
+    WRITE(*,'(a)') 'Modulus-based euclidean algorithm result:'
+    WRITE(*, '(g0)') euclid_mod(a, b)
+    
+    a = 128 * 12
+    b = 128 * 77
+    
+    WRITE(*,'(a)') '[#]'
+    WRITE(*,'(a)') 'Subtraction-based euclidean algorithm result::'
```
Comment (leios) [inline contents/euclidean_algorithm/code/fortran/euclidean.f90:52]:
```suggestion
    WRITE(*,'(a)') 'Subtraction-based euclidean algorithm result:'
```

File: contents/euclidean_algorithm/code/fortran/euclidean.f90 [hunk 3]
```diff
@@ -38,12 +38,18 @@ PROGRAM euclidean
...
+    WRITE(*, '(g0)') euclid_sub(a, b)
```
Comment (leios) [inline contents/euclidean_algorithm/code/fortran/euclidean.f90:53]: Did we just not run the fortran code before?

**Review by leios — APPROVED** [review]
I will approve this, but it does need to be merged up to master. There were a few small changes to the C and scheme code, but they are easy fixes

File: contents/euclidean_algorithm/code/scala/euclidean.scala (hunk 1 above)
Comment (stormofice) [reply contents/euclidean_algorithm/code/scala/euclidean.scala:6]: It didn't.
The scala and javascript implementation needed some (small) fixes to work.

File: contents/euclidean_algorithm/code/fortran/euclidean.f90 (hunk 3 above)
Comment (stormofice) [reply contents/euclidean_algorithm/code/fortran/euclidean.f90:53]: We ran the two methods the other way around

**Comment by Amaras** [comment]
Since implementation and review of the missing languages don't seem to be planned any time soon, I'll just merge this as long as everything compiles correctly.

**Pull request merged**